packages/api/lib/granules.js

```diff
@@ -62,10 +64,37 @@ function getGranuleProductVolume(granuleFiles = []) {
     .reduce((x, y) => x + y, 0);
 }
 
+// TODO default arguments
+const deleteGranuleAndFiles = async (
+  knex,
+  dynamoGranule,
+  pgGranule,
+  filePgModel,
+  granulePgModel,
+  granuleModelClient
+) => {
+  const files = await knex(filePgModel.tableName)
+    .where({ granule_cumulus_id: pgGranule.cumulus_id });
+
+  await knex.transaction(async (trx) => {
```

Comment:
This logic looks good, although I wonder what the performance hit is on this approach over something like a granule delete with cascade and then a parallel S3 object delete..... but that's probably academic given the usual granule->files counts.

Reply:
It's a good question. Something else I've been thinking about -- are we going to run into trouble with nested transactions like this and a Knex/PG connection limit? That's configurable but I'm not sure what this will look like with a large set. Bulk delete might answer that question (which is next on my list in a separate PR).

Reply:
That's a legit concern - one of the cost drivers of our cloud databases are total connections used.  I *think* knex is handling this in a sane way, but we should do some manual testing when we get to the bulk delete PR.

Reply:
The most recent refactor at least partially addressed the performance/connection concern here:

d7230248ee1b906511c141188c56e31173326976

I'll still need to add tests around large numbers of granules/files in the bulk-delete PR but we're eliminating the nested transaction which was a red flag.

Reply:
Great, thanks.